src/source.js

```diff
@@ -186,19 +185,29 @@ class BlockedSource {
         // we keep that as a promise in 'blockRequests' to allow waiting on
         // a single block.
         for (let i = 0; i < group.length; ++i) {
-          const id = group[i];
+          const { id } = group[i];
           this.blockRequests.set(id, (async () => {
-            const response = await request;
-            const o = i * this.blockSize;
-            const t = Math.min(o + this.blockSize, response.data.byteLength);
-            const data = response.data.slice(o, t);
-            this.blockRequests.delete(id);
-            this.blocks.set(id, {
-              data,
-              offset: response.offset + o,
-              length: data.byteLength,
-              top: response.offset + t,
-            });
+            try {
+              const response = await request;
+              const o = i * this.blockSize;
+              const t = Math.min(o + this.blockSize, response.data.byteLength);
+              const data = response.data.slice(o, t);
+              this.blockRequests.delete(id);
+              this.blocks.set(id, {
+                data,
+                offset: response.offset + o,
+                length: data.byteLength,
+                top: response.offset + t,
+              });
+            }
+            catch (err) {
+              if (err.name === 'AbortError') {
+                this.blocks.delete(id);
+                this.blockRequests.delete(id);
+              } else {
+                throw (err);
+              }
```

Comment:
Instead of checking for `err.name`, I think you can just do
```js
try {
} catch (AbortError) {
}
```
Any error that's not an `AbortError` will still raise an exception

Reply:
My apologies, I was thrown off by:
<img width="756" alt="Screen Shot 2020-10-14 at 7 18 32 PM" src="https://user-images.githubusercontent.com/43999641/96055439-15284580-0e52-11eb-96e0-ba16cb3ff62c.png">
See: https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal, but it seems to be working now with catching the `AbortError` directly.    Thanks!

Reply:
Oh I missed that. I'm not sure which is preferable

Reply:
It seems to be working for now, but there might be a backwards compatibility issue now that I think about it....I am also unsure.  Maybe I'll try downgrading FF tomorrow and see what happens.

Reply:
I think I'm going to revert just to be safe - they use my previous syntax here: https://javascript.info/fetch-abort

Reply:
Unfortunately that does not work in JavaScript:
```javascript
try {
} catch (AbortError) {
}
```
As you can see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch), you can only have one `catch` clause, and need to discern the type of the exception inside of it. (Unlike in other languages like C++, Java or Python).

So when you do it with `try`/`catch`, then it would have to look like that:
```javascript
try {
} catch (err) {
    if (err instanceof AbortError) {
        //...
    } else {
        throw err;
    }
}
```

Reply:
@constantinius I don't this works either, I have tried.  I get an "AbortError is not defined" error, likely for a similar reason to the above.

Reply:
Ah, that is because `AbortError` is not an error class, but only the [error name](https://developer.mozilla.org/en-US/docs/Web/API/DOMException#Error_names), so I think your original implementation was correct.